packages/amplify-util-uibuilder/src/commands/utils/createUiBuilderComponent.ts

```diff
@@ -1,25 +1,33 @@
-import { StudioTemplateRendererManager, StudioTemplateRendererFactory, StudioComponent } from '@aws-amplify/codegen-ui';
+import { StudioTemplateRendererManager, StudioTemplateRendererFactory, StudioTemplateRenderer, FrameworkOutputManager, RenderTextComponentResponse } from '@aws-amplify/codegen-ui-old';
 import {
   AmplifyRenderer,
   ReactThemeStudioTemplateRenderer,
-  ReactIndexStudioTemplateRenderer,
   ModuleKind,
   ScriptTarget,
   ScriptKind,
-} from '@aws-amplify/codegen-ui-react';
+} from '@aws-amplify/codegen-ui-react-old';
+import {StudioComponent as StudioComponentNew, StudioTheme} from '@aws-amplify/codegen-ui-new';
 import { getUiBuilderComponentsPath } from './getUiBuilderComponentsPath';
 import { printer } from 'amplify-prompts';
 import { $TSContext } from 'amplify-cli-core';
+import { createUiBuilderComponent as createUiBuilderComponentNew, generateAmplifyUiBuilderIndexFile as generateAmplifyUiBuilderIndexFileNew} from './createUiBuilderComponentNew';
 const config = {
   module: ModuleKind.ES2020,
   target: ScriptTarget.ES2020,
   script: ScriptKind.JSX,
   renderTypeDeclarations: true,
 };
 
-export const createUiBuilderComponent = (context: $TSContext, schema: any) => {
+const isUpdatedSchema = (schema: StudioComponentNew) => {
+  return schema.schemaVersion && schema.schemaVersion == '1.0'
+}
+
+export const createUiBuilderComponent = (context: $TSContext, schema: StudioComponentNew) => {
+  if (isUpdatedSchema(schema)) {
+    return createUiBuilderComponentNew(context, schema);
+  }
   const uiBuilderComponentsPath = getUiBuilderComponentsPath(context);
-  const rendererFactory = new StudioTemplateRendererFactory((component: StudioComponent) => new AmplifyRenderer(component, config));
+  const rendererFactory = new StudioTemplateRendererFactory((component: StudioComponentNew) => new AmplifyRenderer(component, config) as unknown as StudioTemplateRenderer<unknown, StudioComponentNew, FrameworkOutputManager<unknown>, RenderTextComponentResponse>);
```

Comment:
Can this type be named better 'StudioComponentNew'?

Reply:
I'm not too worried about it since it's temporary, unless you have a strong opinion about a new name